drivers/volume/portworx/portworx.go

```diff
@@ -150,6 +154,78 @@ func (d *portworx) CleanupVolume(name string) error {
 	return nil
 }
 
+func (d *portworx) GetStorageDevices(n node.Node) ([]string, error) {
+	const (
+		storageInfoKey = "STORAGE-INFO"
+		resourcesKey   = "Resources"
+		pathKey        = "path"
+	)
+	pxNode, err := d.getClusterManager().Inspect(n.Name)
+	if err != nil {
+		return []string{}, err
+	}
+	devPaths := []string{}
+
+	storageInfo, ok := pxNode.NodeData[storageInfoKey]
+	if !ok {
+		return []string{}, fmt.Errorf("Unable to find storage info for node: %v", n.Name)
+	}
+	storageInfoMap := storageInfo.(map[string]interface{})
+
+	resourcesMapIntf, ok := storageInfoMap[resourcesKey]
+	if !ok || resourcesMapIntf == nil {
+		return []string{}, fmt.Errorf("Unable to find resource info for node: %v", n.Name)
+	}
+	resourcesMap := resourcesMapIntf.(map[string]interface{})
+
+	for _, v := range resourcesMap {
```

Comment:
nit: can we move the devPaths slice above this for loop from line 167.. it seems out of place there

Reply:
moved devPaths down in the function